Loop/Managers/DeviceDataManager.swift

```diff
@@ -1424,8 +1424,63 @@ extension DeviceDataManager {
                 }
             }
         }
-        // Wait up to 25 seconds for uploads triggered by these commands to finish
-        let _ = remoteDataServicesManager.waitForUploadsToFinish(timeout: .now() + TimeInterval(25))
+    }
+    
+    func activateRemoteOverride(_ remoteOverride: TemporaryScheduleOverride?) {
+        loopManager.mutateSettings { settings in settings.scheduleOverride = remoteOverride }
+        self.triggerBackgroundUpload(for: .overrides)
+    }
+    
+    func addRemoteCarbEntry(_ carbEntry: NewCarbEntry, completion: @escaping (_ result: CarbStoreResult<StoredCarbEntry>) -> Void) {
+
```

Comment:
It may look weird at first to see local optional variable for the `UIBackgroundTaskIdentifier`. That was due to the block capturing the local variable. Rob Napier has a good [SO](https://stackoverflow.com/questions/28993628/how-to-pass-background-task-identifier-to-completion-handler) answer on someone who asked about this.